quill-core/src/test/scala/io/getquill/PackageSpec.scala

```diff
@@ -50,6 +50,25 @@ class PackageSpec extends Spec {
     }
   }
 
+  "unquotes duble quotations" in {
+    val q: Quoted[EntityQuery[TestEntity]] = quote {
+      quote(query[TestEntity])
+    }
+    q.toString mustEqual "query[TestEntity]"
+  }
+
+  "unquotes quoted function bodies automatically" in {
+    implicit class QueryOps[Q <: Query[_]](q: Q) {
+      def allowFiltering = quote(infix"$q ALLOW FILTERING".as[Q])
+    }
+
+    val q: Quoted[Int => EntityQuery[TestEntity]] = quote {
```

Comment:
Right, but as far as users are concerned they don't have to specify the return type. That's all I was pointing out.